src/Language/Fixpoint/Solver/EnvironmentReduction.hs

```diff
@@ -568,58 +571,77 @@ mergeDuplicatedBindings xs =
 -- | Inlines some of the bindings whose symbol satisfies a given predicate.
 --
 -- Only works if the bindings don't form cycles.
-substBindings
-  :: Lens' v SortedReft
+substBindingsSimplifyingWith
+  :: (Expr -> Expr)
+  -> Lens' v SortedReft
   -> (Symbol -> Bool)
   -> HashMap Symbol v
   -> HashMap Symbol v
-substBindings vLens p env =
+substBindingsSimplifyingWith simplifier vLens p env =
     -- Circular program here. This should terminate as long as the
     -- bindings introduced by ANF don't form cycles.
-    let env' = HashMap.map (vLens %~ inlineInSortedReft (srLookup filteredEnv)) env
+    let env' = HashMap.map (vLens %~ inlineInSortedReftSimplifyingWith simplifier (srLookup filteredEnv)) env
```

Comment:
Sure.  The reason I did it this way was to preserve the code meaning if the function is used elsewhere, which it is in Prettify.  This would mean changing three call sites, changing the behavior of `inlineInSortedReft` (to not simplify), but remove two non-exported function definitions.

I normally would prefer to preserve the meaning of exported functions unless I have a good reason to, and the functions introduced don't escape the file, but I think I'll defer to you in this case as I don't know the priorities of the project well enough to really make a case.

Will change.